hoard/sources/whoas.py

```diff
@@ -41,29 +43,25 @@ def __next__(self) -> Dataset:
                 continue
             else:
                 dataset = create_from_whoas_dim_xml(record, self.client)
+                if not all([dataset.title, dataset.authors, dataset.description]):
```

Comment:
These are all required fields in our model. It shouldn't be possible to create a `Dataset` if these are empty. If it is we need to change our model, because this is clearly a business rule. It should throw an exception if any of these are empty. We should also probably wrap the dataset creation in a try/except block and deal with that, because otherwise a single invalid record will stop the harvest.